### PR TITLE
User model

### PR DESCRIPTION
**Pull request by Spikey21**
added user model

### CODE REVIEW

File: citizen_report/authSystem/models.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    EMAIL_FIELD = 'email'
+    REQUIRED_FIELDS = ['username',]
+
+    objects = UserManager()
+
+    def __str__(self) -> str:
+        return f'{self.email}'
+
+    def get_absolute_url(self):
+        return "/authSystem/%i/" % (self.pk)
```
Comment (adam-harmasz) [inline citizen_report/authSystem/models.py:91]: tutaj raczej skorzystamy z reverse

Comment (Spikey21) [reply citizen_report/authSystem/models.py]: poprawione w tym commicie
https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/ea7a6c0abee8c72e86f72a6ef8eb48b6494d0ea7

File: citizen_report/citizen_report/settings.py [hunk 2]
```diff
@@ -41,6 +41,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'authSystem'
```
[the remark this answers is not in the document]
Comment (Spikey21) [reply citizen_report/citizen_report/settings.py]: poprawione w tym commicie
https://github.com/SDA-wrzesien-2023-gr2/citizen-report/commit/ea7a6c0abee8c72e86f72a6ef8eb48b6494d0ea7

File: citizen_report/authSystem/forms.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+
+from .models import User
```
[the remark this answers is not in the document]
Comment (Spikey21) [reply citizen_report/authSystem/forms.py]: poprawione w tym commicie
https://github.com/SDA-wrzesien-2023-gr2/citizen-report/commit/ea7a6c0abee8c72e86f72a6ef8eb48b6494d0ea7

File: citizen_report/authSystem/models.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+class User(AbstractBaseUser, PermissionsMixin):
+    class Department(models.TextChoices):
+        APPLICANT = 'APP', _('Applicant'),
+        SYSTEM_SUPERVISOR = 'SYS', _('System supervisor'),
+        ROADS_AND_BRIDGES = 'RBR', _('Roads and bridges'),
+        SEWER_AND_WATERWORKS = 'SWS', _('Sewer and waterworks'),
+        POWER_SUPPLY = 'POW', _('Power supply'),
+        GASWORKS = 'GAS', _('Gasworks'),
+        TELECOMUNICATION = 'TEL', _('Telecommunication'),
```
Comment (adam-harmasz) [inline citizen_report/authSystem/models.py:65]: chyba dwa `M`

File: citizen_report/authSystem/models.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    REQUIRED_FIELDS = ['email',]
+
+    objects = UserManager()
+
+    def __str__(self) -> str:
+        return f'Email: {self.email} ' \
+               f'Username: {self.username}'
+
+    def get_absolute_url(self):
+        return reverse('SignUp',args=self.pk)
```
Comment (adam-harmasz) [inline citizen_report/authSystem/models.py:94]: pep8

File: citizen_report/authSystem/models.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        PUBLIC_SAFETY = 'SAF', _('Public safety')
+
+    email = models.EmailField(max_length=100, unique=True)
+    username = models.CharField(max_length=100, unique=True)
+    is_staff = models.BooleanField(default=False)
+    is_superuser = models.BooleanField(default=False)
+    is_active = models.BooleanField(default=True)
+    department = models.CharField(
+        max_length=100,
+        choices=Department.choices, default=Department.APPLICANT
```
Comment (adam-harmasz) [inline citizen_report/authSystem/models.py:79]: pep8

File: citizen_report/authSystem/urls.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+from django.urls import path
+from django.contrib.auth import views as auth_views
+from django.contrib.auth.decorators import login_required
+from . import views
```
Comment (adam-harmasz) [inline citizen_report/authSystem/urls.py:4]: pep8

Comment (Spikey21) [reply citizen_report/authSystem/urls.py]: https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/ab92b66491a51a17c80c7fd61a68c7e709f5b4b4

File: citizen_report/authSystem/models.py (hunk 6 above)
Comment (Spikey21) [reply citizen_report/authSystem/models.py]: https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/ab92b66491a51a17c80c7fd61a68c7e709f5b4b4

File: citizen_report/authSystem/models.py (hunk 5 above)
Comment (Spikey21) [reply citizen_report/authSystem/models.py]: https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/ab92b66491a51a17c80c7fd61a68c7e709f5b4b4

File: citizen_report/authSystem/models.py (hunk 4 above)
Comment (Spikey21) [reply citizen_report/authSystem/models.py]: https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/ab92b66491a51a17c80c7fd61a68c7e709f5b4b4

File: citizen_report/authSystem/models.py (hunk 6 above)
Comment (Spikey21) [reply citizen_report/authSystem/models.py]: https://github.com/SDA-wrzesien-2023-gr2/citizen-report/pull/3/commits/eb9cbaebc8acb32c227d213c0822f0a5b2b0337a

**Review by adam-harmasz — APPROVED** [review]

**Pull request merged**